### DISCUSSION TITLE
Curious about using one I2S port for input and output (i2s-i2s example)

### DISCUSSION THREAD

**Discussion by yujonglee** [pschatzmann/arduino-audio-tools#851] [Q&A] [answered]
I read this two example. [streams-i2s-i2s](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-i2s-i2s), [streams-i2s-i2s-2](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-i2s-i2s-2).

> This means that the WS and BCK pins are shared.

I am curious what is implication of this. Further more, what is possible disadvantages of using a single port?

Somethings in my mind:
- Need to same I2S config
- Can have low data throughput

It will be very helpful if anyone let me know some possible problems with using one port for both input and output.

Thank you very much.

**Comment by pschatzmann** [comment, marked as answer]
I am not aware of any disadvantages: The only reason for using 2 ports would be if you need to use a different sampling rate or different bits_per sample. 
The solution with one single port is e.g. used in all ESP32 based audio boards: the AudioKit or Lyrat,...
